### DISCUSSION TITLE
If I want to change a collectors code, and work with it, what is the best way to setup a project

### DISCUSSION THREAD

**Discussion by ggoldber** [open-telemetry/opentelemetry-collector#10432] [Q&A]
If I want to change a collectors code, and work with it, what is the best way to setup a project that will pick my own changes, instead of the one in the official repo?

**Comment by Rohlik** [comment]
You can do a fork https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#fork, then do your changes.
You will need a few tools, list here - https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#required-tools
